packages/themed/src/config/makeStyles.ts

```diff
@@ -30,9 +30,7 @@ export const makeStyles =
 
     return useMemo(() => {
       const css =
-        typeof styles === 'function'
-          ? styles({ colors: theme.colors, mode: theme.mode }, props)
```

Comment:
when making a custom theme, the theme shape could look like, for example:

```
colors: {...default RNE colors},
mode: {...default RNE mode},
myCustomColors: {
  text: {
    primary: string;
    secondary: string;
  }
}
```

when consuming `makeStyles` for a component, the current implementation of the `styles()` here ignores any custom fields that have been extended from the `FullTheme`.

This change proposes returning the full `theme` so that users have access to their custom/extended themes.

Reply:
We can extend the `Colors` itself, IMO, having complete theme object is not required.
Use `never` to remove existing colors if they aren't required.

Reply:
Can you explain further how this would work? 

I ask because when we define an extended theme (as per https://github.com/react-native-elements/react-native-elements/pull/3450), in our new `themed.d.ts` file, the custom "theme" name is not a known name (unless RNE adds a specific field to be used for this purpose), which is extended off of `FullTheme`:

```
declare module '@rneui/themed' {

  export interface FullTheme {
    colors: RecursivePartial<Colors>;
    fluidColors: ColorShape;
  }

}
```

As you can see below, `makeStyles()` does not return the full theme since it only returns `theme.colors, mode: theme.mode` and is missing `theme.fluidColors`

![image](https://user-images.githubusercontent.com/15162169/165558321-fff00c09-96a2-4acf-8413-ce01e75c035c.png)

Reply:
Can you try https://github.com/react-native-elements/react-native-elements/issues/3472#issuecomment-1110752617 ?

Reply:
https://github.com/react-native-elements/react-native-elements/issues/3472 solves the issue.  thank you